src/opts.rs

```diff
@@ -111,6 +111,41 @@ pub unsafe fn set_cache_object_limit(kind: ObjectType, size: libc::size_t) -> Re
     Ok(())
 }
 
+/// Set the maximum total data size that will be cached in memory across all
+/// repositories before libgit2 starts evicting objects from the cache. This
+/// is a soft limit, in that the library might briefly exceed it, but will start
+/// aggressively evicting objects from cache when that happens. The default
+/// cache size is 256MB.
+///
+/// # Safety
+/// This function is modifying a C global without synchronization, so it is not
+/// thread safe, and should only be called before any thread is spawned.
+pub unsafe fn set_cache_max_size(size: libc::size_t) -> Result<(), Error> {
```

Comment:
Shouldn't all of these values be `ssize_t` (two s's)?

Not that it makes a huge difference, and it's not clear why it is signed, but probably best to match upstream, no?